tools/editor/scene_tree_dock.cpp

```diff
@@ -1877,6 +1895,13 @@ void SceneTreeDock::_bind_methods() {
 	ObjectTypeDB::bind_method(_MD("instance"),&SceneTreeDock::instance);
 }
 
+SceneTreeDock::NodeNameCasing SceneTreeDock::_get_node_name_casing(const String& p_property) {
+	switch (EditorSettings::get_singleton()->get(p_property).operator int()) {
+		case 1: return NODE_NAME_CASING_CAMEL_CASE;
+		case 2: return NODE_NAME_CASING_SNAKE_CASE;
+	}
+	return NODE_NAME_CASING_PASCAL_CASE;
+}
```

Comment:
You can just do `return (SceneTreeDock::NodeNameCasing) (int) EditorSettings::get_singleton()->get(p_property)` already.